django/db/models/functions/math.py

```diff
@@ -9,6 +9,20 @@
 from django.db.models.lookups import Transform
 
 
+class HyperbolicFallbackMixin:
+
+    def as_mysql(self, compiler, connection, **extra_context):
+        return super().as_sql(compiler, connection, template=self.fallback, **extra_context)
+
+    def as_oracle(self, compiler, connection, **extra_context):
+        extra = {} if not self.is_inverse else {'template': self.fallback}
+        return super().as_sql(compiler, connection, **extra, **extra_context)
+
+    def as_postgresql(self, compiler, connection, **extra_context):
+        extra = {} if connection.features.is_postgresql_12 else {'template': self.fallback}
```

Comment:
This is not a good feature name 😉

Reply:
True. Perhaps we could have a set of supported hyperbolic functions to check whether the fallback is required. Oracle natively supports `cosh`, `sinh`, and `tanh`, but not the inverse functions.